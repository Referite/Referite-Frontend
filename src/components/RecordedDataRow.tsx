import '../styles/RecordedDataRow.css';

interface RecordedDataRowProps {
    country: string;
    goldValue: number;
    silverValue: number;
    bronzeValue: number;
  }
  
  const RecordedDataRow: React.FC<RecordedDataRowProps> = ({ country, goldValue, silverValue, bronzeValue  }) => {

    const getMedalValueClass = (value: number, type: 'gold' | 'silver' | 'bronze') => {
      const baseClass = "medal-value";
      const typeClass = `${type}-medal-value`;
      const boldClass = value !== 0 ? "bold" : "";
      return `${baseClass} ${typeClass} ${boldClass}`;
    }

    return (
      <>
        <div className="recorded-country">{country}</div>
        <div className={getMedalValueClass(goldValue, 'gold')}>{goldValue}</div>
        <div className={getMedalValueClass(silverValue, 'silver')}>{silverValue}</div>
        <div className={getMedalValueClass(bronzeValue, 'bronze')}>{bronzeValue}</div>
        <label> &nbsp; </label>
      </>
    );
  };
  
  export default RecordedDataRow;
  