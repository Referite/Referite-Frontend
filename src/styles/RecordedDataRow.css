.medal-value {
    color: #c9c7c6;
    max-height: 6vh;
    text-align: center;
    font-size: 2.5vw;
    font-family: 'TH SarabunPSK';
    margin-bottom: 7%;
    width: 15%;
  }
  
  .bold {
    font-weight: bold; 
    color: #5a5855;
  }

.silver-medal-value {
    margin-left: 34%;
}

.bronze-medal-value {
    margin-left: 81%;
}

.recorded-country {
    box-sizing: border-box;
    height: 3vw;
    font-size: 2vw;
    font-family: 'TH SarabunPSK';
    justify-content: space-between;
    margin-bottom: 7%;
    font-weight: bold;
    color: #5a5855;
    width: 100%;
    width:19vw;
}
