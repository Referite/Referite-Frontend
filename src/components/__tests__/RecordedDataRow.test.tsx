import renderer from 'react-test-renderer';
import RecordedDataRow from '../RecordedDataRow';

jest.mock('../RecordedDataRow', () => 'RecordedDataRow');
describe('RecordedDataRow Component', () => {
    it('renders correctly', () => {
      const countriesLst = ['Country1', 'Country2'];
      const onButtonClick = jest.fn();
      const onCountrySelect = jest.fn();
      const id = '1';
      const medalValues = {
          input1: '1',
          input2: '0',
          input3: '0'
      };
      const onMedalInputChange = jest.fn();
      const selectedCountry = 'Country1';
      const isNewRowAdded = false;

      const component = renderer.create(
        <RecordedDataRow
          countriesLst={countriesLst}
          onButtonClick={onButtonClick}
          onCountrySelect={onCountrySelect}
          id={id}
          medalValues={medalValues}
          onMedalInputChange={onMedalInputChange}
          selectedCountry={selectedCountry}
          isNewRowAdded={isNewRowAdded}
        />
      );
  
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });