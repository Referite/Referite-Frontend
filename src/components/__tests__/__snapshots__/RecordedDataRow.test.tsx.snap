// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordedDataRow Component renders correctly 1`] = `
<RecordedDataRow
  countriesLst={
    [
      "Country1",
      "Country2",
    ]
  }
  id="1"
  isNewRowAdded={false}
  medalValues={
    {
      "input1": "1",
      "input2": "0",
      "input3": "0",
    }
  }
  onButtonClick={[MockFunction]}
  onCountrySelect={[MockFunction]}
  onMedalInputChange={[MockFunction]}
  selectedCountry="Country1"
/>
`;
